WordPress/Classes/ViewRelated/Cells/MediaSizeSliderCell.swift

```diff
@@ -0,0 +1,143 @@
+import Foundation
+import UIKit
+
+class MediaSizeSliderCell: UITableViewCell {
+    // MARK: - Default values
+    struct Default {
+        static let value = 1500
+        static let minValue = 300
+        static let maxValue = 3000
+        static let step = 100
+
+        static var model: Model {
+            get {
+                return Model(value: value, minValue: minValue, maxValue: maxValue, step: step)
+            }
+        }
+    }
+
+    static let height = 108.0
+
+    // MARK: - Public interface
+    var value: Int {
+        get {
+            return model.value
+        }
+        set {
+            model.value = newValue
+        }
+    }
+
+    var minValue: Int {
+        get {
+            return model.minValue
+        }
+        set {
+            model.minValue = newValue
+        }
+    }
+
+    var maxValue: Int {
+        get {
+            return model.maxValue
+        }
+        set {
+            model.maxValue = newValue
+        }
+    }
+
+    var step: Int {
+        get {
+            return model.step
+        }
+        set {
+            model.step = newValue
+        }
+    }
+
+    var title: String? {
+        get {
+            return titleLabel.text
+        }
+        set {
+            titleLabel.text = newValue
+        }
+    }
+
+    var onChange: (Int -> Void)?
+
+    // MARK: - Private properties
+    private var model: Model = Default.model {
+        didSet {
+            updateSubviews()
+        }
+    }
+
+    func updateSubviews() {
+        slider.minimumValue = model.sliderMinimumValue
+        slider.maximumValue = model.sliderMaximumValue
+        slider.value = model.sliderValue
+        slider.accessibilityValue = model.accessibleText
+        valueLabel.text = model.valueText
+    }
+
+    // MARK: - UIKit bindings
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        updateSubviews()
+    }
+
+    @IBOutlet var titleLabel: UILabel!
+    @IBOutlet var valueLabel: UILabel!
+    @IBOutlet var slider: UISlider!
+
+    @IBAction func sliderChanged(sender: UISlider) {
+        model.value = Int(sender.value)
+        onChange?(model.value)
+    }
+
+    // MARK: - View Model
+    struct Model {
+        var value: Int {
+            didSet {
+                if step > 1 {
+                    value = value
+                        .round(UInt(step))
+                        .clamp(min: minValue, max: maxValue)
+                }
+            }
+        }
+        var minValue: Int
+        var maxValue: Int
+        var step: Int
+
+        var valueText: String {
+            if value == maxValue {
+                return NSLocalizedString("Original", comment: "Indicates an image will use its original size when uploaded.")
+            }
+            let format = NSLocalizedString("%dx%dpx", comment: "Max image size in pixels (e.g. 300x300px)")
+            return NSString(format: format, value, value) as String
```

Comment:
Awesome, I guess autocomplete failed me this time since I think I tried that first